docs/dev/adding_instruments.rst

```diff
@@ -732,6 +644,102 @@ In the above example, :code:`MultimeterA` and :code:`MultimeterB` use a differen
+
+Now to :class:`~pymeasure.instruments.Instrument`. The most important methods are :meth:`~pymeasure.instruments.Instrument.write` and :meth:`~pymeasure.instruments.Instrument.read`, as they are the most basic building blocks for the communication. The pymeasure properties (:meth:`Instrument.control <pymeasure.instruments.Instrument.control>` and its derivatives :meth:`Instrument.measurement <pymeasure.instruments.Instrument.measurement>` and :meth:`Instrument.setting <pymeasure.instruments.Instrument.setting>`) and probably most of your methods and properties will call them. In any instrument, :meth:`write` should write a general string command to the device in such a way, that it understands it. Similarly, :meth:`read` should return a string in a general fashion in order to process it further.
+
+The getter of :meth:`Instrument.control <pymeasure.instruments.Instrument.control>` does not call them directly, but via a chain of methods. It calls :meth:`~pymeasure.instruments.Instrument.values` which in turn calls :meth:`~pymeasure.instruments.Instrument.ask` and processes the returned string into understandable values. :meth:`~pymeasure.instruments.Instrument.ask` sends the readout command via :meth:`write`, waits some time if necessary via :meth:`wait_till_read`, and reads the device response via :meth:`read`.
+
+Similarly, :meth:`Instrument.binary_values <pymeasure.instruments.Instrument.binary_values>` sends a command via :meth:`write`, waits with :meth:`wait_till_read`, but reads the response via :meth:`Adapter.read_binary_values <pymeasure.adapters.Adapter.read_binary_values>`.
+
+
+Adding a device address
+***********************
+
+Let's look at a simple example for a device, which requires its address as the first three characters and returns the same style. This is straightforward, as :meth:`write` just prepends the device address to the command, and :meth:`read` has to strip it again doing some error checking. Similarly, a checksum could be added.
+
+.. code-block:: python
+
+    class Extreme5002(Instrument):
+        """The great Extreme5002 instrument.
+        :param address: The device address for the communication.
+        """
+        def __init__(self, adapter, address=0):
+            super().__init__(adapter, "Extreme5002")
+            self.address = f"{address:03}"
+
+        def write(self, command):
+            """Add the device address in front of every command and send it to the device."""
+            super().write(self.address + command)
+
+        def read(self):
+            """Read from the device and assert, that the response starts with the device address."""
+            got = super().read()
+            if got.startswith(self.address):
+                return got[3:]
+            else:
+                raise ConnectionError(f"Message {got} for wrong address read.")
```

Comment:
```suggestion
                raise ConnectionError(f"Expected message address '{address}', but read message '{got}'.")
```
Always good to delineate where some inserted/quoted strings start and end. Different structure maybe an improvement.